doozer/doozerlib/backend/konflux_client.py

```diff
@@ -712,3 +714,84 @@ def _inner():
+        """
+        if overall_timeout_timedelta is None:
+            overall_timeout_timedelta = datetime.timedelta(hours=5)
+
+        namespace = namespace or self.default_namespace
+        api = await self._get_api(API_VERSION, KIND_RELEASE)
+
+        if self.dry_run:
+            await asyncio.sleep(3)
+            release = {
+                "metadata": {"name": release_name, "namespace": namespace},
+                "apiVersion": API_VERSION,
+                "kind": KIND_RELEASE,
+                "status": {"conditions": [{"type": "Released", "status": "True", "reason": "Succeeded"}]}
+            }
+            self._logger.info(f"[DRY RUN] Would have waited for Release {release_name} to complete")
+            return resource.ResourceInstance(self.dyn_client, release)
+
+        def _inner():
+            nonlocal overall_timeout_timedelta
+            watcher = watch.Watch()
+            released_status = "Not Found"
+            released_reason = "Not Found"
+            timeout_datetime = datetime.datetime.now() + overall_timeout_timedelta
+
+            while True:
+                try:
+                    for event in watcher.stream(
+                        api.get,
+                        resource_version=0,
+                        namespace=namespace,
+                        serialize=False,
+                        field_selector=f"metadata.name={release_name}",
+                        timeout_seconds=5 * 60
+                    ):
+                        assert isinstance(event, Dict)
+                        obj = resource.ResourceInstance(api, event["object"])
+                        # status takes some time to appear
+                        try:
+                            released_condition = art_util.KubeCondition.find_condition(obj, 'Released')
+                            if released_condition:
+                                released_status = released_condition.status
+                                released_reason = released_condition.reason
+                        except AttributeError:
+                            pass
+
+                        self._logger.info("Release %s [status=%s][reason=%s]", release_name, released_status, released_reason)
+
+                        if released_reason not in ["Unknown", "Not Found", "Progressing"]:
+                            watcher.stop()
+                            return obj
+
+                        if datetime.datetime.now() > timeout_datetime:
+                            self._logger.info("Timeout reached. Exiting..")
+                            watcher.stop()
+                            return obj
+
+                    self._logger.info("No updates for Release %s during watch timeout period; requerying",
+                                      release_name)
+                except TimeoutError:
+                    self._logger.error("Timeout waiting for Release %s to complete", release_name)
+                    continue
```

Comment:
If we `continue` on TimeoutError the while loop will restart, and this is true until `datetime.datetime.now() > timeout_datetime:`. In this case we log `Timeout reached. Exiting..` and return. So wondering, what's the point of logging an error here?

Reply:
https://github.com/kubernetes-client/python/blob/master/examples/watch/timeout-settings.md#sever-side-timeout-kwargstimeout_seconds--n 

In case there is a server side error, it will be useful to know